client/src/js/components/Comments/Comment.js

```diff
@@ -33,20 +34,28 @@ export class Comment extends React.Component {
 	onSubmit = e =>
 	{
 		e.preventDefault();
-		const data = {
-			user: this.props.name,
-			id: this.props.id,
-			product: this.props.prodId,
-			text: this.state.body,
-			rating: this.state.rating
+		if(this.props.auth.isAuthenticated)
+		{
+			const data = {
+				user: this.props.auth.user.name,
+				id: this.props.auth.user.id,
+				product: this.props.prodId,
+				text: this.state.body,
+				rating: this.state.rating
+			}
+			this.props.createComment(data);	
+			window.location.reload(false);	
+		}
+		else
+		{
+			this.setState({ error: "You must be logged in to comment."})
 		}
-		this.props.createComment(data);	
-		window.location.reload(false);	
 	}
 	render()
 	{
 		return(
 			<form noValidate onSubmit={this.onSubmit}>
+				<span>{this.state.error}</span>

```

Comment:
Looks good, guess you can also add in red text for better UI like:
```suggestion
				<span style="color:red">{this.state.error}</span>
```

or maybe wrap the span using styled components like we've been doing